### PR TITLE
update: Event Planners Temp Blips CMD

### LINKED ISSUE #601: Allow event planners to add temporary blips on map for events

### PR DESCRIPTION
**Pull request by Savage912-Official**
Temporary Blips for event planners.

closes #601

Previews:
image
image

### CODE REVIEW

**Comment by minipunch** [comment]
Does this work even if the player relogs?

**Comment by Savage912-Official** [comment]
> Does this work even if the player relogs?
> 
> If not, then u might want to move more of the logic to the server side so it can handle syncing when player's load in and what not better

True I’ll work on it Monday

**Comment by K7AKEN** [comment]
@Savage912-Official asked me to work on this for him, Is there any other way to commit my changes? as it blocks me from doing so, should I put it in another pr, or comment it below?

**Comment by Savage912-Official** [comment]
> @Savage912-Official asked me to work on this for him, Is there any other way to commit my changes? as it blocks me from doing so, should I put it in another pr, or comment it below?

Added you as a contributor on my repo

**Comment by minipunch** [comment]
FYI I saw there were changes committed by @K7AKEN but it tells me this about them:

> "This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository."

So If those changes were made to K7's fork of the repo then u need to make a PR from there to Savage's fork (with the same branch as this PR) and then it should show up here

**Comment by K7AKEN** [comment]
> FYI I saw there were changes committed by @K7AKEN but it tells me this about them:
> 
> > "This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository."
> 
> So If those changes were made to K7's fork of the repo then u need to make a PR from there to Savage's fork (with the same branch as this PR) and then it should show up here

That's weird because they show up on his branch/fork as well, I'll check it out more when I get home.

**Comment by minipunch** [comment]
> > FYI I saw there were changes committed by @K7AKEN but it tells me this about them:
> > > "This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository."
> > 
> > 
> > So If those changes were made to K7's fork of the repo then u need to make a PR from there to Savage's fork (with the same branch as this PR) and then it should show up here
> 
> That's weird because they show up on his branch/fork as well, I'll check it out more when I get home.

Hm wait I think it did apply the changes here actually judging by the commit history so nevermind that 👌

**Comment by Savage912-Official** [comment]
Any updates?

**Comment by minipunch** [comment]
Gonna try it soon

**Review by minipunch — CHANGES_REQUESTED** [review]
See comments below + also I relogged and the blip I placed still went away

**Comment by minipunch** [comment]
Did u make sure the blip didn't go away when u relogged? Cause it does for me as I stated above @K7AKEN

**Comment by Savage912-Official** [comment]
Gonna start working on it in a bit

File: resources/usa_eventPlanners/cl_eventPlanners.lua [hunk 1]
```diff
@@ -19,4 +21,65 @@ Citizen.CreateThread(function()
...
+end)
+
+RegisterNetEvent('usa_eventPlanners:client:blipDialog')
+AddEventHandler('usa_eventPlanners:client:blipDialog', function()
+    
+    local input = lib.inputDialog('Temporary Blip Information', {
+        {type = 'input', label = 'Coordinates', description = 'Format: x, y, z'},
+        {type = 'input', label = 'Name', description = 'Blip Name Shown On Map'},
+        {type = 'number', label = 'Duration (In Minutes)', description = 'How long the blip should be on the map'},
+        {type = 'input', label = 'Icon', description = 'Icon of the blip (Reference list you were given.)'},
```
[the remark this answers is not in the document]
Comment (Savage912-Official) [reply resources/usa_eventPlanners/cl_eventPlanners.lua:33]: I was thinking we could put the link in event planners chat for it so people could grab it from there.

File: resources/usa_eventPlanners/cl_eventPlanners.lua [hunk 2]
```diff
@@ -19,4 +21,65 @@ Citizen.CreateThread(function()
         end
         Wait(1)
     end
+end)
+
+RegisterNetEvent('usa_eventPlanners:client:blipDialog')
+AddEventHandler('usa_eventPlanners:client:blipDialog', function()
+    
+    local input = lib.inputDialog('Temporary Blip Information', {
+        {type = 'input', label = 'Coordinates', description = 'Format: x, y, z'},
```
[the remark this answers is not in the document]
Comment (Savage912-Official) [reply resources/usa_eventPlanners/cl_eventPlanners.lua]: Done

**Comment by Savage912-Official** [comment]
> Did u make sure the blip didn't go away when u relogged? Cause it does for me as I stated above @K7AKEN

I just tested and the blip stayed when I relogged.

**Comment by K7AKEN** [comment]
> > Did u make sure the blip didn't go away when u relogged? Cause it does for me as I stated above @K7AKEN
> 
> I just tested and the blip stayed when I relogged.

Same here, it worked perfectly for me. Only thing is that it says blip even if there isn't one. And the blip list you mentioned

**Review by minipunch — APPROVED** [review]

**Comment by minipunch** [comment]
Nice, it does seem to work actually, good stuff!

**Pull request merged**